satpy/readers/ami_l1b.py

```diff
@@ -201,11 +207,12 @@ def get_dataset(self, dataset_id, ds_info):
         data = data.where(qf == 0)
 
         # Calibration values from file, fall back to built-in if unavailable
-        gain = self.nc.attrs["DN_to_Radiance_Gain"]
-        offset = self.nc.attrs["DN_to_Radiance_Offset"]
+        self.gain = self.nc.attrs["DN_to_Radiance_Gain"]
+        self.offset = self.nc.attrs["DN_to_Radiance_Offset"]
```

Comment:
Why `self.` versus just passing them to `_clip_negative_radiance`?

Additionally, it is considered bad practice to create instance attributes that are not defined at creation time (`__init__`). It isn't obvious to me though why these should be instance attributes though.

Reply:
Fixed!